packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,363 @@
+        PlacementStrategy: noEmpty(flatten((this.options?.placementStrategies ?? []).map((c) => c.toJson().map(uppercaseKeys)))),
+      },
+    };
+
+    function uppercaseKeys(obj: { [key: string]: any }): { [key: string]: any } {
+      const ret: { [key: string]: any } = {};
+      for (const key of Object.keys(obj)) {
+        ret[key.slice(0, 1).toUpperCase() + key.slice(1)] = obj[key];
+      }
+      return ret;
+    }
+
+    function flatten<A>(xs: A[][]): A[] {
+      return Array.prototype.concat([], ...xs);
+    }
+
+    function noEmpty<A>(xs: A[]): A[] | undefined {
+      if (xs.length === 0) {
+        return undefined;
+      }
+      return xs;
+    }
+  }
+}
+
+/**
+ * Run a Task on ECS or Fargate
+ */
+export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  /**
+   * Manage allowed network traffic for this service
+   */
+  public readonly connections: ec2.Connections = new ec2.Connections();
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private securityGroup?: ec2.ISecurityGroup;
+  private networkConfiguration?: any;
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: EcsRunTaskProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EcsRunTask.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN && !sfn.FieldUtils.containsTaskToken(props.containerOverrides)) {
+      throw new Error('Task Token is required in `containerOverrides` for callback. Use Context.taskToken to set the token.');
+    }
+
+    if (!this.props.taskDefinition.defaultContainer) {
+      throw new Error('A TaskDefinition must have at least one essential container');
+    }
+
+    if (this.props.taskDefinition.networkMode === ecs.NetworkMode.AWS_VPC) {
+      this.configureAwsVpcNetworking();
+    } else {
+      // Either None, Bridge or Host networking. Copy SecurityGroup from ASG.
+      this.validateNoNetworkingProps();
+      this.connections.addSecurityGroup(...this.props.cluster.connections.securityGroups);
+    }
+
+    for (const override of this.props.containerOverrides ?? []) {
+      const name = override.containerDefinition.containerName;
+      if (!cdk.Token.isUnresolved(name)) {
+        const cont = this.props.taskDefinition.node.tryFindChild(name);
+        if (!cont) {
+          throw new Error(`Overrides mention container with name '${name}', but no such container in task definition`);
+        }
+      }
+    }
```

Comment:
All this can be dropped in favour of -

```ts
taskDefinition.containers.includes(override.containerDefinition)
```

Reply:
I thought so too, but `containers` is a protected property and not public